### PR TITLE
Ticket 9 post comment

### PR DESCRIPTION
**Pull request by adnaani**
9. Post a new comment to an existing review

### CODE REVIEW

**Review by leekli — CHANGES_REQUESTED** [review]
Hi Adnaan, a good start to the ticket however there are a few things to note/changes requested:

* The comment form requires me to enter a username. There should either be a hard coded user already set up for this, or have a 'logged in' user if this has been created (think about that useContext we covered last week on how to achieve this). We shouldn't expect the user to have to manually type in their username to submit a comment, there's also nothing to tell the user which usernames they can use if they were to type a username in. This app is an app you will be showing to prospective employers, so you want to make it as easy as possible for them to navigate it and add comments.
* I can see it doesn't allow me submit a new comment if the comment box is empty, but it doesn't give me, as a user, any error message to tell me the field is required... as a user it just looks like it doesn't work when I try to submit an empty comment. Have a look at using something like the 'required' tag in the <textarea> element.
* On the new comment form, the 'write' button doesn't seem as obvious to me, as a user, as it could be. Can you maybe change this to 'submit', or 'post'. I'd also look into turning it into something which more visually represents a button, currently it just looks like text and isn't obvious to me as a user as being something i should click.
* When I submitted a new comment, the Chrome console presented me with an error message about each child element not having a unique key prop - take a look.

Please review these changes, implement them and then submit a new pull request once complete.

File: src/components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
 import { Link } from "react-router-dom";
 import { SidebarData } from "./SidebarData";
 import "./Navbar.css";
+import { UserContext } from "../../context/userContext";
 
 const Navbar = () => {
   const [sidebar, setSidebar] = useState(false);
-
+  const { loggedUser } = useContext(UserContext);
+  console.log(loggedUser);
```
Comment (Rosem10) [inline src/components/Navbar/Navbar.jsx]: Please remove!

File: src/utils/dateCalculator.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export const dateCalculator = (date) => {
+  const now = new Date(Date.now());
+  const then = new Date(date).valueOf();
+
+  const age = Math.floor((now - then) / 60000);
+
+  switch (true) {
```
Comment (Rosem10) [inline src/utils/dateCalculator.js:7]: This is lovely.

**Review by Rosem10 — CHANGES_REQUESTED** [review]
All in all really good. The only change I'd like you to make is the behaviour once the user has posted a comment. Currently, the user gets visual feedback in the form of the comment being posted at the bottom of the list - however if there are a lot of comments, the user would need to scroll right to the bottom for confirmation, which isn't great UX. I'd suggest either having some sort of message/ alert pop up once the comment is posted, or having the comment appear at the top of the list, so the user gets instant gratification.

**Pull request merged**